agent/api/task_unix.go

```diff
@@ -15,16 +15,126 @@
 
 package api
 
+import (
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
 const (
 	portBindingHostIP = "0.0.0.0"
+	maxTaskVCPULimit  = 10     // 10 VCPUs limit
+	defaultCPUPeriod  = 100000 // 100ms
+	// Reference: http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDefinition.html
+	minimumCPUShare = 2
 )
 
-func (task *Task) adjustForPlatform() {}
+func (task *Task) adjustForPlatform(cfg *config.Config) {
+	task.memoryCPULimitsEnabledLock.Lock()
+	defer task.memoryCPULimitsEnabledLock.Unlock()
+	task.memoryCPULimitsEnabled = cfg.TaskCPUMemLimit
+}
 
 func getCanonicalPath(path string) string { return path }
 
-// CgroupEnabled checks whether cgroups need be enabled at the task level
-func (task *Task) CgroupEnabled() bool {
-	//stub for now
-	return false
+// BuildCgroupRoot helps build the task cgroup prefix
+// Example: /ecs/task-id
+func (task *Task) BuildCgroupRoot() (string, error) {
+	taskID, err := task.GetID()
+	if err != nil {
+		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id from taskARN: %s", task.Arn)
+	}
+
+	cgroupRoot := filepath.Join(config.DefaultTaskCgroupPrefix, taskID)
```

Comment:
minor: you can combine these to just one line: `return filepath.Join(...), nil`

Reply:
Sure